internal/host/host.go

```diff
@@ -419,3 +414,66 @@ func (m *Manager) reportInstallationMetrics(ctx context.Context, h *models.Host,
 		m.metricApi.ReportHostInstallationMetrics(log, cluster.OpenshiftVersion, h, previousProgress, CurrentStage)
 	}
 }
+
+func (m *Manager) AutoAssignRole(ctx context.Context, h *models.Host, db *gorm.DB) error {
+	// select role if needed
+	if h.Role == models.HostRoleAutoAssign {
+		return m.autoRoleSelection(ctx, h)
+	}
+	return nil
+}
+
+func (m *Manager) selectRole(ctx context.Context, h *models.Host) (models.HostRole, error) {
+	var (
+		autoSelectedRole = models.HostRoleWorker
+		log              = logutil.FromContext(ctx, m.log)
+	)
+
+	// count already existing masters
+	mastersCount := 0
+	if err := m.db.Model(&models.Host{}).Where("cluster_id = ? and status != ? and role = ?",
```

Comment:
There are cases that a host does not have sufficient hardware to be even a worker.  How is that taken into account?

Reply:
It will be marked as insufficient worker.

Reply:
OK